src/shared_library.cc

```diff
@@ -242,4 +242,92 @@ SharedLibrary::GetEntrypoint(
   return Status::Success;
 }
 
+Status
+SharedLibrary::AddAdditionalDependencyDir(
+    const std::string& additional_path, std::wstring& original_path)
+{
+#ifdef _WIN32
+  const std::wstring PATH(L"Path");
+
+  DWORD len = GetEnvironmentVariableW(PATH.c_str(), NULL, 0);
+  if (len > 0) {
+    original_path.resize(len);
+    GetEnvironmentVariableW(PATH.c_str(), &original_path[0], len);
+  } else {
+    return Status(Status::Code::INTERNAL, "PATH variable is empty");
+  }
+
+  LOG_VERBOSE(1) << "Environment before extending PATH: "
+                 << std::string(original_path.begin(), original_path.end());
+
+  std::wstring updated_path_value =
+      std::wstring(additional_path.begin(), additional_path.end());
+  updated_path_value += original_path;
```

Comment:
You don't need to add delimiter for concatenation?

Reply:
The assumption I've made is that users will add a `;` to each path they add at the command line--much like they would adding any other path. However, I believe we can be ultra-conservative here and add a `;` regardless. I believe any extra `;` will be ignored anyways.

Reply:
Reflecting on this more: We can technically add `;` to help the user (though this would only help if they specified a single path). However, I don't want to help/encourage the user to use the extension improperly. I think the best thing to do here would probably be to throw an error if their input string does not end with `;`.